operational/azure/vms_without_managed_disks/azure_vms_without_managed_disks.pt

```diff
@@ -0,0 +1,147 @@
+end
+
+parameter "param_azure_tenant_id" do
+  type "string"
+  label "Azure AD Tenant ID"
+  category "Azure"
+end
+
+parameter "param_azure_sub" do
+  type "string"
+  label "Azure Subscription ID"
+  category "Azure"
+end
+
+###############################################################################
+# Authentication
+###############################################################################
+
+auth "azure_auth", type: "oauth2" do
+  token_url join(["https://login.microsoftonline.com/",$param_azure_tenant_id,"/oauth2/token"])
+  grant type: "client_credentials" do
+    client_id cred("AZURE_APPLICATION_ID")
+    client_secret cred("AZURE_APPLICATION_KEY")
+    additional_params do {
+      "resource" => "https://management.azure.com/"
+    } end
+  end
+end
+
+###############################################################################
+# Pagination
+###############################################################################
+
+pagination "azure_pagination" do
+  get_page_marker do
+    body_path "nextLink"
+  end
+  set_page_marker do
+    uri true
+  end
+end
+
+###############################################################################
+# Datasources
+###############################################################################
+
+datasource "ds_azure_virtual_machines" do
+  request do
+    auth $azure_auth
+    pagination $azure_pagination
+    host "management.azure.com"
+    path join(["/subscriptions/", $param_azure_sub, "/providers/Microsoft.Compute/virtualMachines"])
+    query "api-version","2018-06-01"
+    header "User-Agent", "RS Policies"
+  end
+  result do
+    encoding "json"
+    collect jmes_path(response, "value[*]") do
+      field "id", jmes_path(col_item,"id")
+      field "type", jmes_path(col_item,"type")
+      field "name", jmes_path(col_item,"name")
+      field "osDisk", jmes_path(col_item,"properties.storageProfile.osDisk")
+    end
+  end
+end
+
+datasource "ds_filtered_virtual_machines" do
+  run_script $js_filter_virtual_machines, $ds_azure_virtual_machines
+end
+
+###############################################################################
+# Scripts
+###############################################################################
+script "js_filter_virtual_machines", type: "javascript" do
+  parameters "ds_azure_virtual_machines"
+  result "results"
+  code <<-EOS
+    var results = [];
+    _.each(ds_azure_virtual_machines, function(vm){
+      if (vm.osDisk.vhd != null && vm.osDisk.vhd !== undefined){
+        results.push(vm)
+      }
+    })
+EOS
+end
+###############################################################################
+# Policy
+###############################################################################
+
+policy "azure_resource_policy" do
+  validate_each $ds_filtered_virtual_machines do
+    summary_template "{{ rs_project_name }} (Account ID: {{ rs_project_id }}): {{ len data }} Azure Resources out of compliance"
```

Comment:
This is a cloud native policy, we shouldn't include RS Project Name and ID in the summary template.